src/merge_iterator.rs

```diff
@@ -150,11 +189,18 @@ impl<T: KeyValueIterator> PartialOrd<Self> for MergeIteratorHeapEntry<T> {
 
 impl<T: KeyValueIterator> Ord for MergeIteratorHeapEntry<T> {
     fn cmp(&self, other: &Self) -> Ordering {
-        match self.next_kv.key.cmp(&other.next_kv.key) {
-            Ordering::Less => Ordering::Less,
-            Ordering::Greater => Ordering::Greater,
-            Ordering::Equal => self.index.cmp(&other.index),
-        }
+        // the row entry with higher index, should always has higher seqnum
+        // if a key both exists in L0_1 and L0_2, the one in L0_2 should be
+        // returned first, as L0_2 is created later than L0_1, and the seqnum
+        // is higher.
+        //
+        // we'll wrap a Reverse in the BinaryHeap, so the cmp here is in
+        // increasing order.
+        (&self.next_kv.key, self.index, self.next_kv.seq).cmp(&(
```

Comment:
this will yield the keys from lowest sequence number to highest sequence number. I think that's fine, though suboptimal, since the advance fns loop over all keys with a given sequence number. We could invert that by writing an explicit comparator for the sequence number that orders higher sequence numbers before lower ones.